### ISSUE TITLE
Plugin dependencies resolution causes unreported incompatibilites between plugins

### ISSUE THREAD

**Issue by amisevsk** [eclipse-che/che#15794]
### Describe the bug
Starting a workspace with multiple plugins, where one plugin depends on the other (e.g. the `vscode-kubernetes` plugin, which depends on `vscode-yaml`), causes one of the plugins to fail to initialize (most of the time). 

This appears to be due to functionality in upstream Theia [1] that automatically downloads any `.vsix` dependencies in a plugin before initializing -- e.g. when starting a workspace with the kubernetes plugin, I have
```
$ ls /tmp/vscode-unpacked/
ms-kubernetes-tools.vscode-kubernetes-tools.latest.mhdmaprors.vscode-kubernetes-tools-1.0.4.vsix
redhat_vscode_yaml.vsix
```
in the kubernetes-sidecar container.

This means:
1. Workspaces with seemingly unsatisfied dependencies work without issue (a workspace with just the kubernetes plugin will download vscode-yaml on start and function correctly)
2. Workspaces where plugin dependencies are explicitly included do not work. 

[1] - https://github.com/eclipse-theia/theia/commit/2d4c8457c09cc6ca37fc5480d14eb080a74ed85a

### Che version
 - [x] nightly

### Steps to reproduce
Start a workspace from the devfile
```yaml
apiVersion: 1.0.0
metadata:
  name: plugin-conflict
components:
  - id: ms-kubernetes-tools/vscode-kubernetes-tools/latest
    type: chePlugin
  - id: redhat/vscode-yaml/latest
    type: chePlugin
```
Note error toast notification on start:
```
Cannot activate the 'Kubernetes' extension because it depends on the 'redhat.vscode-yaml' extension, which is not installed.
```
and kubernetes plugin output:
```
LOG:PLUGIN_HOST(1): PluginManagerExtImpl/init()
LOG:PLUGIN_HOST(1): initializing(vscode-kubernetes-tools@1.0.4 with /remote-endpoint/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init.js)
LOG:Debugger contribution has been registered: draft
WARN:Unsupported activation events: onFileSystem:k8smsx, please open an issue: https://github.com/eclipse-theia/theia/issues/new
WARN:ms-kubernetes-tools.vscode-kubernetes-tools extension will be activated eagerly.
WARN:Cannot activate the 'Kubernetes' extension because it depends on the 'redhat.vscode-yaml' extension, which is not installed.
LOG:PLUGIN_HOST(1): PluginManagerExtImpl/init()
ERROR:Failed to initialize redhat.vscode-yaml plugin.Error: ENOENT: no such file or directory, open '/tmp/vscode-unpacked/redhat.vscode-yaml.latest.uqipzeaahj.redhat.vscode-yaml-0.4.0.vsix/extension/package.json'
```
Note also that the failed plugin depends on order in the devfile, if `redhat/vscode-yaml/latest` is placed first, it will fail instead with output:
```
ERROR:Failed to initialize redhat.vscode-yaml plugin.Error: ENOENT: no such file or directory, open '/tmp/vscode-unpacked/redhat_vscode_yaml.vsix/extension/package.json'
ERROR:Failed to initialize ms-kubernetes-tools.vscode-kubernetes-tools plugin.Error: ENOENT: no such file or directory, open '/tmp/vscode-unpacked/ms-kubernetes-tools.vscode-kubernetes-tools.latest.mhdmaprors.vscode-kubernetes-tools-1.0.4.vsix/extension/package.json'
```

Further, the kubernetes plugin starts without issue when there is no conflicting plugin (remove vscode-yaml and restart workspace)

### Additional context
Potentially blocks https://github.com/eclipse/che/issues/15376

Related issue: https://github.com/eclipse/che/issues/15730

**Comment by amisevsk** [comment]
I'm going to limit the scope of this issue to https://github.com/eclipse/che/pull/16046; while the plugin dependencies problem is larger than that PR, there are other issues (e.g. https://github.com/eclipse/che/issues/15376, https://github.com/eclipse/che/issues/15966, https://github.com/eclipse/che/issues/15844) to address those problems.

**Comment by amisevsk** [comment]
Fixed by #16046, https://github.com/eclipse/che-plugin-broker/pull/93
